modules/objdetect/src/qrcode.cpp

```diff
@@ -2301,6 +2349,17 @@ inline Point computeOffset(const vector<Point>& v)
 bool QRDecode::versionDefinition()
 {
     CV_TRACE_FUNCTION();
+    CV_LOG_INFO(NULL, "QR corners: " << original_points[0] << " " << original_points[1] << " " << original_points[2] <<
+                      " " << original_points[3]);
+    double numModules = getNumModules();
+    const double version_tmp = (numModules - 21.) * .25 + 1.;
+    if (abs(version_tmp - cvRound(version_tmp)) < 0.16 && cvRound(version_tmp) < 7 && cvRound(version_tmp) > 0) {
```

Comment:
I propose to extract 0.16 as constant and add comment about meaning and value origin.